python/example_code/neptune/analytics/neptune_analytics_query_example.py

```diff
@@ -0,0 +1,71 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import boto3
+from botocore.exceptions import ClientError
+
+# snippet-start:[neptune.python.graph.execute.main]
+"""
+Running this example.
+
+----------------------------------------------------------------------------------
+VPC Networking Requirement:
+----------------------------------------------------------------------------------
+Amazon Neptune must be accessed from **within the same VPC** as the Neptune cluster.
+It does not expose a public endpoint, so this code must be executed from:
+
+  - An **AWS Lambda function** configured to run inside the same VPC
+  - An **EC2 instance** or **ECS task** running in the same VPC
+  - A connected environment such as a **VPN**, **AWS Direct Connect**, or a **peered VPC**
+
+"""
+
+NEPTUNE_ANALYTICS_ENDPOINT = "https://<your-neptune-analytics-endpoint>:8182"
+GRAPH_ID = "<your-graph-id>"
+
+def main():
+    # Build the boto3 client for neptune-graph with endpoint override
+    client = boto3.client(
+        "neptune-graph",
+        endpoint_url=NEPTUNE_ANALYTICS_ENDPOINT
+    )
+
+    try:
+        execute_gremlin_profile_query(client, GRAPH_ID)
+    except Exception as e:
+        print(f"Unexpected error in main: {e}")
+
+def execute_gremlin_profile_query(client, graph_id):
+    """
+    Executes a Gremlin or OpenCypher query on Neptune Analytics graph.
+
+    Args:
+        client (boto3.client): The NeptuneGraph client.
+        graph_id (str): The graph identifier.
+    """
+    print("Running openCypher query on Neptune Analytics...")
+
+    try:
+        response = client.execute_query(
+            GraphIdentifier=graph_id,
```

Comment:
The capitalization in the parameters of all of these methods does not match the [documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/neptune-graph/client/execute_query.html) and it also does not match the code provided by the SME:

```
`resp =client.execute_query(
    graphIdentifier = graph_id,
    queryString="MATCH (n {code: 'ANC'}) RETURN n",
    language='OPEN_CYPHER'
)

print(resp['payload'].read().decode('UTF-8')`
```

Please recheck that the parameters match exactly on all these graph and data examples.

Reply:
Fixed